budgetportal/webflow/views.py

```diff
@@ -60,12 +67,73 @@ def provincial_infrastructure_project_detail(request, id, slug):
         % (snapshot.name, snapshot.province),
         "page_description": "Provincial infrastructure project by the %s %s department."
         % (snapshot.province, snapshot.department),
+        "csv_download_url": project.csv_download_url,
     }
     return render(
         request, "webflow/detail_provincial-infrastructure-projects.html", context
     )
 
 
+class ProvInfraProjectCSVGeneratorMixIn:
+    labels = {
+        "adjustment_appropriation_professional_fees": "adjusted_appropriation_professional_fees",
+        "adjustment_appropriation_construction_costs": "adjusted_appropriation_construction_costs",
+        "adjustment_appropriation_total": "adjusted_appropriation_total",
+        "total_project_cost": "estimated_total_project_cost",
+    }
+
+    def generate_csv_response(self, response_results, filename="export.csv"):
+        response = StreamingHttpResponse(
+            streaming_content=self._generate_rows(response_results),
+            content_type="text/csv",
+        )
+        response["Content-Disposition"] = 'attachment; filename="{}"'.format(filename)
+        return response
+
+    def _generate_rows(self, response_results):
+        headers = self._get_headers(ProvInfraProjectCSVSerializer.Meta.fields)
+        writer = csv.DictWriter(Echo(), fieldnames=headers)
+        yield writer.writerow({h: h for h in headers})
+
+        for row in response_results:
+            prepared_row = self._prepare_row(row)
+            yield writer.writerow(prepared_row)
+
+    def _get_headers(self, headers):
+        return [self.labels.get(header, header) for header in headers]
```

Comment:
if you have time, fixing this field name in the model (mapping at import-time) and index would be preferable